log/wrapper.go

```diff
@@ -272,6 +273,36 @@ func ErrorWithSentryWrapper() Wrapper {
 	}
 }
 
+// PrometheusCounterWrapper returns a Wrapper implementation that increases
+// counter by 1 then calls delegate to log the message.
+//
+// Please note that it's not possible to deserialize this Wrapper directly from
+// yaml, so you usually need to override it in your main function, after
+// baseplate.ParseConfigYAML call, for example:
+//
+//     // a global variable
+//     var tracingFailures = promauto.NewCounterVec(prometheus.CounterOpts{
+//       Namespace: "myservice",
+//       Subsystem: "tracing",
+//       Name:      "failures_total",
+//       Help:      "Total number of failures when sending tracing spans to the sidecar",
+//     }, nil /* no labels */).With(nil /* no labels */)
```

Comment:
Don't use a Vec if you don't have labels.

If you're going to have an example with labels, the With call should happen at the PrometheusCounterWrapper call-site.